flux-middle/src/fhir.rs

```diff
@@ -91,12 +91,36 @@ pub struct SortDecl {
 
 pub type SortDecls = FxHashMap<Symbol, SortDecl>;
 
+#[derive(Debug, Clone)]
+pub struct GenericPredicates {
+    pub parent: Option<DefId>,
+    pub predicates: Vec<ClauseKind>,
+}
```

Comment:
hmm yes, I thought about that. IIRC think the reason I put it here, in `fhir` post-desugaring, is that the `conv` step is not supposed to throw errors, so I thought better to put it in the `desugar` since if you e.g. cannot resolve the `Item` then that will be too "late" in `conv` ? (Or am I getting mixed up?)

If we do move it, just so I follow you suggest that `fhir::ClauseKind` be replaced with something like

```rust
pub struct WhereBoundPredicate {
    pub span: Span,
    pub bounded_ty: Ty,
    pub bounds: Vec<Path>,
}
```

where `Path` is as your comment above?

Reply:
Hmm that's a good point. Conv doesn't fail right now because everything is guaranteed to succeed by a previous pass (e.g. well-formenmess) or by the rust type checker. I'm too far removed from the code right now to know if we can assume that. I'll let you decide. I don't think is too important now, we can refactor later

Reply:
I think the absence of errors in this case will be guaranteed by checking that the predicates are a correct refinement of the rust predicates (during `annot_check`) which I think we are not checking now.

Reply:
> If we do move it, just so I follow you suggest that `fhir::ClauseKind` be replaced with something like
> 
> ```rust
> pub struct WhereBoundPredicate {
>     pub span: Span,
>     pub bounded_ty: Ty,
>     pub bounds: Vec<Path>,
> }
> ```
> 
> where `Path` is as your comment above?

 Also yes to this